conda_spawn/shell.py

```diff
@@ -164,6 +164,38 @@ def executable(self):
         return "zsh"
 
 
+class ShellShell(PosixShell):
+    Activator = activate.ShellActivator
+
+    def executable(self):
+        return "shell"
+
+    def args(self):
+        return ("--interact", "--norc")
+
+    def spawn_popen(
+        self, command: Iterable[str] | None = None, **kwargs
+    ) -> subprocess.Popen:
+        try:
+            with NamedTemporaryFile(
+                prefix="conda-spawn-",
+                suffix=self.Activator.script_extension,
+                delete=False,
+                mode="w",
+            ) as f:
+                f.write(self.script())
+            return subprocess.Popen(
+                [self.executable(), *self.args(), f.name], env=self.env(), **kwargs
+            )
+        finally:
+            self._files_to_remove.append(f.name)
+
+    def spawn(self, command: Iterable[str] | None = None) -> int:
+        proc = self.spawn_popen(command)
```

Comment:
Can't we use the tty in Unix systems?

Reply:
Ttys are needed for PS1 support I'm afraid.

Reply:
In some shells, the PS1 variable is not ever exported, only local to the current session. So a process like `conda`'s doesn't "see it" when it wants to expand it with something like `PS1 = os.environ.get("PS1, "") + " (base)"`. 

But if Prefix's `shell` always exports it I guess we are ok.